R/preprocess.R

```diff
@@ -193,20 +258,61 @@ enw_new_reports <- function(obs) {
     by = c("group")
   ]
   reports <- reports[, delay := 0:(.N - 1), by = c("reference_date", "group")]
+
+  if (!is.null(reports$max_confirm)) {
+    reports[, prop_reported := new_confirm / max_confirm]
+  }
+
+  if (set_negatives_to_zero) {
+    reports <- reports[new_confirm < 0, new_confirm := 0]
+  }
   return(reports[])
 }
 
-#' @title FUNCTION_TITLE
-#' @description FUNCTION_DESCRIPTION
-#' @param obs PARAM_DESCRIPTION
-#' @return OUTPUT_DESCRIPTION
+#' Filter observations to restrict the maximum reporting delay
+#'
+#' @return A data frame filtered so that dates by report are less than or equal
+#' the reference date plus the maximum delay.
+#'
+#' @inheritParams enw_new_reports
+#' @inheritParams enw_preprocess_data
+#' @family preprocess
+#' @export
+#' @importFrom data.table copy
+#' @examples
+#' obs <- enw_example("preprocessed")$obs[[1]]
+#' enw_filter_obs(obs, max_delay = 2)
+enw_filter_obs <- function(obs, max_delay) {
+  obs <- data.table::copy(obs)
+  obs <- obs[, .SD[report_date <= (reference_date + max_delay - 1)],
```

Comment:
This excludes the maximum delay, unlike stated in the function `@return` description.